wallet/src/components/Purses.tsx

```diff
@@ -1,33 +1,61 @@
 import { useState } from 'react';
-import { CircularProgress } from '@mui/material';
+import ArrowDownward from '@mui/icons-material/ArrowDownward';
+import ArrowUpward from '@mui/icons-material/ArrowUpward';
 import Button from '@mui/material/Button';
-import Transfer from './Transfer';
+import IbcTransfer, { IbcDirection } from './IbcTransfer';
 import PurseAmount from './PurseAmount';
 import { withApplicationContext } from '../contexts/Application';
 import CardItem from './CardItem';
 import Card from './Card';
 import ErrorBoundary from './ErrorBoundary';
 import Loading from './Loading';
+import { ibcAssets } from '../util/ibc-assets';
+import type { PurseInfo } from '../service/Offers';
+import type { KeplrUtils } from '../contexts/Provider';
 
 import './Purses.scss';
+import { agoricChainId } from '../util/ibcTransfer';
+
+interface TransferPurse {
+  purse?: PurseInfo;
+  direction?: IbcDirection;
+}
+
+interface Props {
+  purses: PurseInfo[] | null;
+  previewEnabled: boolean;
```

Comment:
surprised to see this passed along since it's a global state.

Reply:
Passing it as a prop from context allows components to reactively update when you type `setPreviewEnabled(true)` in the console.